lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -2964,11 +2965,35 @@ void ByteCodeGenerator::EmitOneFunction(ParseNode *pnode)
 
         DefineLabels(funcInfo);
 
-        if (pnode->sxFnc.HasNonSimpleParameterList() || !funcInfo->IsBodyAndParamScopeMerged())
+        // We need to emit the storage for special symbols before we emit the default arguments in case the default
+        // argument expressions reference those special names.
+        if (pnode->sxFnc.HasNonSimpleParameterList())
         {
-            Assert(pnode->sxFnc.HasNonSimpleParameterList() || CONFIG_FLAG(ForceSplitScope));
+            // If the param and body scope are merged, the special symbol vars are located in the body scope so we
+            // need to walk over the var list.
+            if (funcInfo->IsBodyAndParamScopeMerged())
+            {
+                for (ParseNodePtr pnodeVar = pnode->sxFnc.pnodeVars; pnodeVar; pnodeVar = pnodeVar->sxVar.pnodeNext)
+                {
+                    Symbol* sym = pnodeVar->sxVar.sym;
 
-            if (paramScope)
+                    if (sym != nullptr && !(pnodeVar->sxVar.isBlockScopeFncDeclVar && sym->GetIsBlockVar()) && sym->IsSpecialSymbol())
```

Comment:
What's the case where one of the special symbols is also the var binding of a block-scoped function?

Reply:
I don't think there is one. This code was lifted from DefineUserVars, which does this check. I'll try to remove it.